challenge-05/challenge-05.js

```diff
@@ -47,30 +67,51 @@ propriedades:
 - Se o parâmetro não for passado, a função deve retornar o objeto com todos
 os livros.
 */
-// ?
+function book(bookName) {
+	var nameBooks = {
+		'name1': {
+			quantidadePaginas: 300,
+			autor: 'George Martin',
+			editora: 'Fantasy'
+		},
+		'name2': {
+			quantidadePaginas: 200,
+			autor: 'Peter M.',
+			editora: 'Devir'		
+		},
+		'name3': {
+			quantidadePaginas: 250,
+			autor: 'Rafael Tavares',
+			editora: 'Metralha Inc.'	
+		}
+	};
+
+	return bookName === undefined ? nameBooks : nameBooks [bookName]; 
```

Comment:
-1
Sempre que for buscar um índice de um array, pode manter "colado" o `[` com o nome do array, ok? :)